src/figuresClasses.ts

```diff
@@ -1,11 +1,119 @@
-export interface Figure {}
+type ShapeType = `triangle` | `circle` | `rectangle`;
+type ColorType = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+function checkSidesLength(...sides: number[]): boolean {
+  for (const side of sides) {
+    if (side <= 0) {
+      return false;
+    }
+  }
 
-export class Circle implements Figure {}
+  return true;
+}
+
+function roundToHundr(n: number): number {
+  return Number(parseInt(`${n * 100}`) / 100);
+}
+
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: ColorType,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const validSidesLength: boolean = checkSidesLength(this.a, this.b, this.c);
+    let validSides: boolean;
+    const maxSide = Math.max(this.a, this.b, this.c);
+
+    switch (maxSide) {
+      case this.a:
+        validSides = maxSide < this.b + this.c;
+        break;
+
+      case this.b:
+        validSides = maxSide < this.a + this.c;
+        break;
+
+      case this.c:
+        validSides = maxSide < this.a + this.b;
+        break;
+    }
+
+    if (!validSidesLength) {
```

Comment:
[CODE STYLE] - When throwing an error, provide a more detailed description to convey the exact reason for the error.